drools-wb-screens/drools-wb-enum-editor/drools-wb-enum-editor-backend/src/test/java/org/drools/workbench/screens/enums/backend/server/indexing/EnumIndexVisitorCDITest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.enums.backend.server.indexing;
+
+import org.assertj.core.api.Assertions;
+import org.guvnor.test.CDITestSetup;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.soup.project.datamodel.oracle.ProjectDataModelOracle;
+import org.uberfire.java.nio.file.Path;
+import org.uberfire.java.nio.file.Paths;
+
+public class EnumIndexVisitorCDITest extends CDITestSetup {
+
+    private EnumFileIndexer indexer;
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        indexer = getReference(EnumFileIndexer.class);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        super.cleanup();
+    }
+
+    @Test
+    public void testDependentEnumIndexing() throws Exception {
+        final Path testedPath = Paths.get(getClass().getResource("cars/src/main/resources/com/myteam/repro/cars.enumeration").toURI());
+        indexer.fillIndexBuilder(testedPath);
+        final ProjectDataModelOracle oracle = indexer.getProjectDataModelOracle(testedPath);
```

Comment:
@jomarko OK, so this can be merged, but this needs updating to reflect the changes made on ```master```